torch/sparse/semi_structured.py

```diff
@@ -0,0 +1,329 @@
+                [1., 1., 1., 1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1., 1., 1., 1.],
+                [1., 1., 1., 1., 1., 1., 1., 1.]], device='cuda:0',
+               dtype=torch.float16))
+
+
+    """
+    warnings.warn(
+        (
+            "The PyTorch API of SparseSemiStructuredTensor is in prototype stage "
+            "and will change in the near future. Please open a Github issue "
+            "for features requests and see our documentation on the torch.sparse "
+            "module for further information about the project."
+        ),
+        UserWarning,
+    )
+    # check device
+    if not original_tensor.is_cuda:
+        raise RuntimeError((f"Error original_tensor.device= {original_tensor.device} is not supported! "
+                            "Only CUDA tensors are currently supported."))
+
+    # check dim
+    if original_tensor.dim() != 2:
+        raise RuntimeError((f"Error original_tensor.dim = {original_tensor.dim()} is not supported! "
+                            "Only 2d tensors are currently supported."))
+
+    # check dtype
+    if original_tensor.dtype not in _DTYPE_TO_CUSPARSELT_CONFIG:
+        raise RuntimeError((f"Error original_tensor.dtype {original_tensor.dtype} is not a supported dtype! "
+                            "dtype must be one of: {_DTYPE_TO_CUSPARSELT_CONFIG}"))
+
+    # check shape
+    m, n = original_tensor.shape
+    min_size = _DTYPE_TO_CUSPARSELT_CONFIG[original_tensor.dtype].min_size
+    if m < min_size or m % min_size or n < min_size or n % min_size:
+        # TODO add padding here
+        raise RuntimeError((f"Error original_tensor.shape {original_tensor.shape} is not supported! "
+                            "Both dimensions must be larger than and a multiple of {min_size}"))
+
+
+    # This code calculates the size of the compressed tensor.
+    # compression factor is different based on dtype
+    num_bytes = original_tensor.nelement() * original_tensor.element_size()
+    compression_factor = _DTYPE_TO_CUSPARSELT_CONFIG[original_tensor.dtype].compression_factor
+    compressed_size_bytes = num_bytes * compression_factor // 16
+    compressed_size = compressed_size_bytes // original_tensor.element_size()
+
+    compressed_tensor = torch.empty(
+        (compressed_size,),
+        dtype=original_tensor.dtype,
+        device=original_tensor.device,
+    )
+
+    # try to use cuSPARSELt
+    # TODO default to CUTLASS is cuSPARELt is not avaliable
+    cslt = torch._C.sparse.cusparselt(compressed_tensor)
```

Comment:
You're probably working on removing this right now. Just pointing it out. Then we can land a CUTLASS-only version for now and focus on the design of the Tensor subclass.